sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -60,6 +60,21 @@ case class BroadcastHashJoinExec(
     }
   }
 
+  override def outputPartitioning: Partitioning = {
+    def buildKeys: Seq[Expression] = buildSide match {
+      case BuildLeft => leftKeys
+      case BuildRight => rightKeys
+    }
+
+    streamedPlan.outputPartitioning match {
+      case h: HashPartitioning =>
+        PartitioningCollection(Seq(h, HashPartitioning(buildKeys, h.numPartitions)))
```

Comment:
When `buildSide` is right and join type is left outer, will we see NULL values on `buildSide` in some partitions? Is `HashPartitioning(buildKeys)` still met at this case?

Reply:
Ah, good point. Do you see any issue if I scope this down to support only the `inner` join type?

Reply:
For inner join, seems okay to me.

Reply:
+1 on the idea limiting this opt. only for inner joins. Could you update the PR description, too?